src/test/java/school/hei/haapi/integration/PaymentIT.java

```diff
@@ -0,0 +1,207 @@
+
+    List<Payment> actual = api.getStudentFeePayments(STUDENT1_ID, FEE1_ID, 1, 5);
+
+    assertTrue(actual.contains(payment1()));
+    assertTrue(actual.contains(payment2()));
+  }
+
+  @Test
+  void student_read_ko() {
+    ApiClient student1Client = anApiClient(STUDENT1_TOKEN);
+    PayingApi api = new PayingApi(student1Client);
+
+    assertThrowsApiException(
+        "{\"type\":\"403 FORBIDDEN\",\"message\":\"Access is denied\"}",
+        () -> api.getStudentFeePayments(STUDENT2_ID, FEE3_ID, null, null));
+  }
+
+  @Test
+  void teacher_read_ko() {
+    ApiClient teacher1Client = anApiClient(TEACHER1_TOKEN);
+    PayingApi api = new PayingApi(teacher1Client);
+
+    assertThrowsApiException(
+        "{\"type\":\"403 FORBIDDEN\",\"message\":\"Access is denied\"}",
+        () -> api.getStudentFeePayments(STUDENT2_ID, FEE3_ID, null, null));
+  }
+
+  @Test
+  void manager_write_ok() throws ApiException {
+    ApiClient manager1Client = anApiClient(MANAGER1_TOKEN);
+    PayingApi api = new PayingApi(manager1Client);
+
+    List<Payment> actual = api.createStudentPayments(STUDENT1_ID, FEE3_ID,
+        List.of(creatablePayment1()));
+
+    List<Payment> expected = api.getStudentFeePayments(STUDENT1_ID, FEE3_ID, 1, 5);
+    assertTrue(expected.containsAll(actual));
+  }
+
+  @Test
+  void teacher_write_ko() {
+    ApiClient teacher1Client = anApiClient(TEACHER1_TOKEN);
+    PayingApi api = new PayingApi(teacher1Client);
+
+    assertThrowsApiException(
+        "{\"type\":\"403 FORBIDDEN\",\"message\":\"Access is denied\"}",
+        () -> api.createStudentPayments(STUDENT1_ID, FEE1_ID, List.of()));
+
+  }
+
+  @Test
+  void student_write_ko() {
+    ApiClient student1Client = anApiClient(STUDENT1_TOKEN);
+    PayingApi api = new PayingApi(student1Client);
+
+    assertThrowsApiException(
+        "{\"type\":\"403 FORBIDDEN\",\"message\":\"Access is denied\"}",
+        () -> api.createStudentPayments(STUDENT1_ID, FEE1_ID, List.of()));
+  }
+
+  @Test
+  void manager_write_ko() throws ApiException {
+    ApiClient manager1Client = anApiClient(MANAGER1_TOKEN);
+    PayingApi api = new PayingApi(manager1Client);
+    List<Payment> expected = api.getStudentFeePayments(STUDENT1_ID, FEE3_ID, 1, 5);
+
+    assertThrowsApiException(
+        "{\"type\":\"400 BAD_REQUEST\",\"message\":\"Payment amount (6000)"
+            + " exceeds fee remaining amount (5000)\"}",
+        () -> api.createStudentPayments(STUDENT1_ID, FEE3_ID, List.of(creatablePayment3())));
+
+    List<Payment> actual = api.getStudentFeePayments(STUDENT1_ID, FEE3_ID, 1, 5);
+    assertEquals(0, expected.size());
+    assertEquals(expected, actual);
+  }
+
+  @Test
+  void manager_write_with_some_bad_fields_ko() {
+    ApiClient manager1Client = anApiClient(MANAGER1_TOKEN);
+    PayingApi api = new PayingApi(manager1Client);
+    CreatePayment toCreate1 = creatablePayment1().amount(null);
+    CreatePayment toCreate2 = creatablePayment1().amount(-1);
```

Comment:
Unless type is FIX! But just create a ticket for addressing FIX types for the moment

Reply:
My above comment is bad. Indeed, amount can NEVER be negative. The FIX type is used to cancel fees that we don't want pay (eg: fees that we created falsly).

Reply:
For sure, but this feature is not still implemented so I've created a ticket anyway : [#60 ](https://github.com/hei-school/hei-admin-api/issues/60)